src/legacy/core_plugins/kibana/public/discover/angular/directives/table.tsx

```diff
@@ -0,0 +1,388 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useMemo, useState, useCallback, ReactNode } from 'react';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiButtonToggle,
+  EuiDataGrid,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutHeader,
+  EuiIcon,
+  EuiLink,
+  EuiPortal,
+  EuiTitle,
+  useRenderToText,
+  htmlIdGenerator,
+  EuiButtonEmpty,
+  EuiSpacer,
+  EuiDataGridColumn,
+  EuiDataGridCellValueElementProps,
+} from '@elastic/eui';
+import { DocViewer } from '../../components/doc_viewer/doc_viewer';
+import { IndexPattern } from '../../kibana_services';
+import {
+  ElasticSearchHit,
+  DocViewFilterFn,
+} from '../../../../../../ui/public/registry/doc_views_types';
+
+type Direction = 'asc' | 'desc';
+type SortArr = [string, Direction];
+interface SortObj {
+  id: string;
+  direction: Direction;
+}
+
+interface Props {
```

Comment:
@kertal Could you check if these types are OK/correct?

Reply:
looks good, just would suggest `ariaLabelledby` -> `ariaLabelledBy` and `filter` -> `onFilter`

Reply:
Searching through the codebase it seems we/EUI leave the aria labels exactly matching the HTML properties so I'm going to go with `aria-labelledby` so that you can pass in the normal HTML property.